backend/api/routes/files.py

```diff
@@ -246,10 +341,15 @@ async def delete_file(
         .select_from(File)
         .filter_by(project_id=file.project_id)
         .filter_by(hash=file.hash)
-        .filter_by(status=FileStatus.LOCAL.value)
     )
     number_of_duplicate_files = session.scalars(stmt).one()
     if number_of_duplicate_files == 1:
-        file_location = file.local_fpath
-        file_location.unlink(missing_ok=True)
+        if file.status == FileStatus.LOCAL:
+            file.local_fpath.unlink(missing_ok=True)
+        if file.status == FileStatus.S3:
```

Comment:
No, the delay should be if FileStatus is `UPLOADING` (ie. we can't delete now because it's not done uploading but dont want it to stay for 7d on the server).
If it's in S3 already, we should enqueue for immediate deletion

Reply:
Done